fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -126,26 +126,37 @@ class FullyShardedDataParallel(nn.Module):
         mixed_precision (bool, Optional):
             if ``True``, inputs, activations and gradients will be kept in FP16;
             computation and communication will occur in FP16; and a (sharded)
-            master copy of the model weights will be maintained in FP32.
-        fp32_reduce_scatter (bool, Optional):
-            if ``True``, then reduce-scatter gradients in FP32. This is only
-            relevant when *``mixed_precision``* is ``True``.
+            master copy of the model weights will be maintained in FP16.
+            Weight, buffer and communication data types can be overriden using
+            the 3 options below.
+            Note, this is expected to work with autocast context from pytorch.
+            Note, some layers, e.g. convolutions, may only work with
+            ``fp32_compute_dtype==True`` below.
+            Default: False
+        fp32_compute_dtype (bool, Optional):
+            if ``True``, use FP32 for full parameters for computation.
+            Default: None, which means default to FP16 or FP32 based on the
+            *``mixed_precision``* argument.
+        fp32_buffer_dtype (bool, Optional):
+            if ``True``, use FP32 for buffers for computation.
+            Default: None, which means using the value of the
+            *``fp32_compute_dtype``* argument.
+        fp32_reduce_scatter (bool):
+            if ``True``, then reduce-scatter gradients in FP32.
+            This is only relevant when *``mixed_precision``* is ``True``.
```

Comment:
I filed an issue to later perhaps decouple this with mixed precision. after ddp has a fp16 grad hook even without mixed precision.